### PR TITLE
Parallel chains and some more fixes

### LINKED ISSUE #307: Question: MCMC being "sticky" or stuck across different seeds
Not sure if this is the best place to post this, but we are running into some scenarios where the posterior distribution is having inconsistent results across different seeds (see attached alpha and rho plots). Looking for any potential recommendations/suggestions!

The data has `71 n_items` (i.e., "companies") with pair-wise rankings and was fitted using both `kendall` and `footrule` distances which produced similar results (we ended up picking kendall to avoid calculating the partition function). We also tried to fine-tune the bmm model using the **following hyper-parameters**, but the stickiness results are still the same (e.g., for company "Genomic Health Inc." in the third row, the posterior is centered at either very left or very right depending on the seed number across different parameter settings).

```
bmm_nmc_values <- c(100000) 
leap_size_values <- c(floor(n_items / 2), floor(n_items / 5), floor(n_items / 7))
alpha_prop_sd_values <- c(0.1, 0.5)
rho_thinning_values <- c(100)
alpha_jump_values <- c(10)
lambda_values <- c(0.001, 0.01, 0.1, 0.2)
bmm_burnin <- bmm_nmc*0.5 
```

We also tried switching `shuffle_unranked` and `random` in `generate_initial_ranking`, but `random=TRUE` takes too long given `71 n_items` and not sure if this could be the root cause. Judging by the alpha trace plot, it also seems like the MCMC has somewhat converged and increasing the number of iterations to `1 million` didn't help as well. Any feedback/suggestions would be greatly appreciated!

S0221_2016_posteriordensity_rho_v_burn_5e+050_seed103_20230903
S0221_2016_posteriordensity_rho_v_burn_5e+050_seed102_20230903
S0221_2016_posteriordensity_rho_v_burn_5e+050_seed101_20230903

S0221_2016_trace_alpha_v0_seed103_20230903
S0221_2016_trace_alpha_v0_seed102_20230903
S0221_2016_trace_alpha_v0_seed101_20230903

### PR DESCRIPTION
**Pull request by osorensen**
Should fix #307, #309, and #321

### CODE REVIEW

**Review by wleoncio — APPROVED** [review]
Good job, really enjoyed seeing a new vignette for parallelization! :)
